src/mediapipe_internal/nodestate.cpp

```diff
@@ -0,0 +1,95 @@
+//*****************************************************************************
+// Copyright 2023 Intel Corporation
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+//*****************************************************************************
+#include "nodestate.hpp"
+
+#include <filesystem>
+#include <string>
+
+#include <spdlog/spdlog.h>
+
+#include "../status.hpp"
+
+#if (PYTHON_DISABLE == 0)
+#include <pybind11/embed.h>  // everything needed for embedding
+
+#include "src/mediapipe_calculators/python_backend_calculator.pb.h"
+#endif
+
+namespace ovms {
+
+NodeState::NodeState() {
+#if (PYTHON_DISABLE == 0)
+    this->pythonNodeState = py::none();
+#endif
+}
+
+NodeState::NodeState(const NodeState& other) {
+#if (PYTHON_DISABLE == 0)
+    this->pythonNodeState = other.pythonNodeState;
+#endif
+}
+
+#if (PYTHON_DISABLE == 0)
+Status NodeState::Create(const google::protobuf::Any node_options) {
+    mediapipe::PythonBackendCalculatorOptions options;
+    node_options.UnpackTo(&options);
+    auto fs_handler_path = std::filesystem::path(options.handler_path());
+    fs_handler_path.replace_extension();
```

Comment:
Why are we removing extension? Shouldn't we error when the path is not pointing towards directory instead?